addons/account/models/account_move.py

```diff
@@ -3564,7 +3601,19 @@ class AccountMoveLine(models.Model):
     _sql_constraints = [
         (
             'check_credit_debit',
-            'CHECK(credit + debit>=0 AND credit * debit=0)',
+            '''CHECK(
+                (
+                    (is_storno = TRUE)
+                    AND
+                    (credit * debit=0)
+                )
+                OR
+                (
+                    (is_storno = FALSE)
+                    AND
+                    (credit + debit>=0 AND credit * debit=0)
```

Comment:
factorize in order to keep the minimal difference related to storno accounting

(C is True & X) | (C is False & (X & Y))
=> X & (C is True | Y)